### PR TITLE
Pin Python core dependencies (virtualenv, pip, setuptools)

### PR DESCRIPTION
**Pull request by humitos**
Pin Python core versions to avoid problems when rebuilding images to release a new patch version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,8 +14,16 @@ ENV PYTHON_VERSION_36 3.6.8
...
+
 # Note: 4.7.12.1 drastically increases memory usage
 ENV CONDA_VERSION 4.6.14
+
+# Use _PIP_VERSION (with starting _) because of a bug in Pip
+# https://github.com/pypa/pip/issues/4528
+ENV _PIP_VERSION 20.0.1
+
+ENV SETUPTOOLS_VERSION 45.1.0
+ENV VIRTUALENV_VERSION 16.7.9
```
Comment (ericholscher) [inline Dockerfile]: I'd like these to go _after_ the pyenv install, so that we don't have to rebuild all python versions to change pip versions.

Comment (ericholscher) [reply Dockerfile]: That can be another PR tho, so we don't block this one.

**Comment by humitos** [comment]
I've merged this into `releases/6.x` (`LABEL version=6.0.3`) to backport this changes.

**Comment by humitos** [comment]
The backported image is building at https://hub.docker.com/repository/registry-1.docker.io/readthedocs/build/builds/10c2c6f2-4824-4eb3-bb57-536757409438

**Pull request merged**